operator/builtin/transformer/move/move_test.go

```diff
@@ -0,0 +1,320 @@
+package move
+
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/testutil"
+)
+
+type processTestCase struct {
+	name      string
+	expectErr bool
+	op        *MoveOperatorConfig
+	input     func() *entry.Entry
+	output    func() *entry.Entry
+}
+
+func TestMoveProcess(t *testing.T) {
+	newTestEntry := func() *entry.Entry {
+		e := entry.New()
+		e.Timestamp = time.Unix(1586632809, 0)
+		e.Body = map[string]interface{}{
+			"key": "val",
+			"nested": map[string]interface{}{
+				"nestedkey": "nestedval",
+			},
+		}
+		return e
+	}
+
+	cases := []processTestCase{
+		{
+			"MoveBodyToBody",
```

Comment:
Can we have a couple tests where the body is entirely overwritten?
1. A test where a string value is moved to `$body`
2. A test where an object is moved to `$body`